chainer/testing/random.py

```diff
@@ -118,38 +117,43 @@ def test_func(self, *args, **kw):
 
         if isinstance(impl, _bundle._ParameterizedTestCaseBundle):
             cases = impl
-        elif isinstance(impl, type) and issubclass(impl, unittest.TestCase):
+        else:
             tup = _bundle._TestCaseTuple(impl, None, None)
             cases = _bundle._ParameterizedTestCaseBundle([tup])
-        else:
-            raise ValueError('Can\'t apply fix_random to {}'.format(impl))
 
-        for case, _, _ in cases.cases:
+        for klass, _, _ in cases.cases:
             # Applied to test case class
-            klass = case
+            if hasattr(klass, 'setUp'):
+                setup_method_name = 'setUp'
+                teardown_method_name = 'tearDown'
+            else:
+                setup_method_name = 'setup'
+                teardown_method_name = 'teardown'
```

Comment:
This kind of module dependency in "reverse" direction does not seem favorable.
By doing this `fix_random` assumes a set of interface requirents on the target, and in this case the requirements is just an instance locally defined in ChainerX tests.
In other words, it's like "`chainer.testing` module depends on a local class in ChainerX tests". 

I'd instead suggest either of the following:
1. Abstract `OpTest`'s interface to make it suitable for dependency from `testing`. I don't think this option is not really appropriate in this case, because there wouldn't be other cases to reuse the interface than `OpTest`.
2. Refactor `fixed_random` so that it has two distinct APIs but share the same implementation. One is `chainer.testing` which currently exists, and the other will be located near `OpTest`.
3. Use different way to fix random seeds. I think this is the easiest option.

Reply:
Okay I agree, changed to 2.